### ISSUE TITLE
Infinite recursion in randn(Int)

### ISSUE THREAD

**Issue by tbreloff** [JuliaLang/julia#17608]
This should throw a DomainError or something similar, but instead:

```
julia> randn(Int)
ERROR: StackOverflowError:
 in randn(::MersenneTwister, ::Type{Int64}) at ./random.jl:1217
 in randn!(::MersenneTwister, ::Array{Int64,0}) at ./random.jl:1208
 in randn(::MersenneTwister, ::Type{Int64}) at ./random.jl:1217
 in randn!(::MersenneTwister, ::Array{Int64,0}) at ./random.jl:1208
 in randn(::MersenneTwister, ::Type{Int64}) at ./random.jl:1217
...
< thousands of similar lines >
```

If I control-C while the error output is printing, I get a crash:

```
 in randn!(::MersenneTwister, ::Array{Int64,0}) at ./random.jl:1208jl_uv_writecb() ERROR: bad file descriptor EBADF
jl_uv_writecb() ERROR: bad file descriptor EBADF
jl_uv_writecb() ERROR: bad file descriptor EBADF
jl_uv_writecb() ERROR: bad file descriptor EBADF
jl_uv_writecb() ERROR: bad file descriptor EBADF
rec_backtrace at /opt/julia5/src/stackwalk.c:84
record_backtrace at /opt/julia5/src/task.c:232
jl_throw at /opt/julia5/src/task.c:550
enq_work at ./event.jl:55
#schedule#257 at ./event.jl:71
jl_call_method_internal at /opt/julia5/src/julia_internal.h:178 [inlined]
jl_apply_generic at /opt/julia5/src/gf.c:2034
#schedule at ./<missing>:0
uv_writecb_task at ./stream.jl:861
uv__write_callbacks at /opt/julia5/deps/srccache/libuv-ecbd6eddfac4940ab8db57c73166a7378563ebd3/src/unix/stream.c:967
uv__stream_destroy at /opt/julia5/deps/srccache/libuv-ecbd6eddfac4940ab8db57c73166a7378563ebd3/src/unix/stream.c:450
uv__finish_close at /opt/julia5/deps/srccache/libuv-ecbd6eddfac4940ab8db57c73166a7378563ebd3/src/unix/core.c:256
uv__run_closing_handles at /opt/julia5/deps/srccache/libuv-ecbd6eddfac4940ab8db57c73166a7378563ebd3/src/unix/core.c:286
uv_run at /opt/julia5/deps/srccache/libuv-ecbd6eddfac4940ab8db57c73166a7378563ebd3/src/unix/core.c:356
jl_atexit_hook at /opt/julia5/src/init.c:313
jl_exit at /opt/julia5/src/jl_uv.c:537
_start at ./client.jl:365
jl_call_method_internal at /opt/julia5/src/julia_internal.h:178 [inlined]
jl_apply_generic at /opt/julia5/src/gf.c:2034
unknown function (ip: 0x40185c)
unknown function (ip: 0x4012f6)
unknown function (ip: 0x401348)
```

**Comment by NHDaly** [comment]
I'm not seeing this at `v0.4.5`. Did you see this when building from `master`?

**Comment by NHDaly** [comment]
To be more clear, this is what I saw in `Julia Version 0.4.5`:

```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.5 (2016-03-18 00:58 UTC)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |  x86_64-apple-darwin14.5.0

julia> randn(Int)
ERROR: MethodError: `randn` has no method matching randn(::Type{Int64})
```

**Comment by simonster** [comment]
I can reproduce this on master.

**Comment by stevengj** [comment]
Closed by #17627
